aws/resource_aws_ec2_transit_gateway_peering_attachment_test.go

```diff
@@ -0,0 +1,413 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_ec2_transit_gateway_peering_attachment", &resource.Sweeper{
+		Name: "aws_ec2_transit_gateway_peering_attachment",
+		F:    testSweepEc2TransitGatewayPeeringAttachments,
+	})
+}
+
+func testSweepEc2TransitGatewayPeeringAttachments(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).ec2conn
+	input := &ec2.DescribeTransitGatewayPeeringAttachmentsInput{}
+	var sweeperErrs *multierror.Error
+
+	for {
+		output, err := conn.DescribeTransitGatewayPeeringAttachments(input)
```

Comment:
Nit: The AWS Go SDK provides a paginated function for this, [`DescribeTransitGatewayPeeringAttachmentsPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeTransitGatewayPeeringAttachmentsPages)

Reply:
04b8a62